types/bull-arena/index.d.ts

```diff
@@ -1,56 +1,61 @@
-// Type definitions for bull-arena 2.6
+// Type definitions for bull-arena 3.0
 // Project: https://github.com/bee-queue/arena/
 // Definitions by: Levi Bostian <https://github.com/levibostian>
+//                 Gaurav Sharma <https://github.com/gtpan77>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 4.1
 
 import { RequestHandler } from "express";
 import { ClientOpts } from "redis";
+import Bull = require("bull");
+import Bee = require("bee-queue");
 
 declare function Arena(
-  options: BullArena.MiddlewareOptions,
-  listenOptions?: BullArena.MiddlewareListenOptions
+    options: BullArena.MiddlewareOptions,
+    listenOptions?: BullArena.MiddlewareListenOptions
 ): RequestHandler;
 
 declare namespace BullArena {
-  interface MiddlewareOptions {
-    queues: Array<QueueOptions & ConnectionOptions>;
-  }
-
-  interface MiddlewareListenOptions {
-    port?: number;
-    host?: string;
-    basePath?: string;
-    disableListen?: boolean;
-    useCdn?: boolean;
-  }
-
-  interface QueueOptions {
-    name: string;
-    hostId?: string;
-    type?: "bull" | "bee";
-    prefix?: "bull" | "bq" | string;
-  }
-
-  type ConnectionOptions =
-    | PortHostConnectionOptions
-    | RedisUrlConnectionOptions
-    | RedisClientConnectionOptions;
-
-  interface PortHostConnectionOptions {
-    host: string;
-    port?: number;
-    password?: string;
-    db?: string;
-  }
-
-  interface RedisUrlConnectionOptions {
-    url: string;
-  }
-
-  interface RedisClientConnectionOptions {
-    redis: ClientOpts;
-  }
+    interface MiddlewareOptions {
+        Bull?: typeof Bull;
+        Bee?: typeof Bee;
```

Comment:
Maybe I don't know enough about Typescript, but I am wondering what is the reason for using `typeof ...` here? Why not use `Bull?: Bull;`?

Reply:
I used `Bull?: Bull;` first. Faced this error `Cannot use namespace 'Bull' as a type.`

Reply:
Ok. I don't know how to fix this. I am not that good with type definitions to know if what you have works or not.

Reply:
How I tested this - Created an sample project and edited index.d.ts file of bull-arena with these changes of given in this comment thread. It is working fine.